go/vt/vtgate/vcursor_impl.go

```diff
@@ -929,6 +930,31 @@ func (vc *vcursorImpl) SetPriority(priority string) {
 	}
 }
 
+func (vc *vcursorImpl) SetExecQueryTimeout(timeout *int) {
+	// Determine the effective timeout: use passed timeout if non-nil, otherwise use session's query timeout if available
+	var execTimeout *int
+	if timeout != nil {
+		execTimeout = timeout
+	} else if sessionTimeout := vc.GetQueryTimeout(0); sessionTimeout > 0 {
+		execTimeout = &sessionTimeout
+	}
+
+	// If no effective timeout and no session options, return early
+	if execTimeout == nil {
+		if vc.safeSession.GetOptions() == nil {
+			return
+		}
+		vc.safeSession.GetOrCreateOptions().Timeout = nil
+		return
+	}
```

Comment:
I have a strong preference to convert this to a `switch` statements with three `case` clauses, and also get rid of the int pointer.
```suggestion
	var execTimeout int
	switch {
	case timeout != nil:
		execTimeout = *timeout
	case sessionTimeout := vc.GetQueryTimeout(0); sessionTimeout > 0:
		execTimeout = sessionTimeout
	default:
		if vc.safeSession.GetOptions() == nil {
				return
		}	
		vc.safeSession.GetOrCreateOptions().Timeout = nil
		return
	}
```

Reply:
We cannot send down `0` if the `QUERY_TIMEOUT_MS` comment directive is not `0`. 

In other ways like the VTGate flag or the system variable, the default is `0` so it cannot be used to mark authoritative vttablet timeout.

Only when `QUERY_TIMEOUT_MS` is set to `0` in the query comment, the authoritative timeout is set to infinite.